giab_runner/giab_runner.py

```diff
@@ -0,0 +1,332 @@
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    first_line = result.stdout.splitlines()[0]
+    commit_hash = first_line.split(" ")[0]
+    return commit_hash
+
+
+def check_valid_repo(logger: Logger, repo: Path):
+    if not repo.exists():
+        logger.error(f'The folder "{repo}" does not exist')
+        sys.exit(1)
+
+    if not repo.is_dir():
+        logger.error(f'"{repo}" is not a folder')
+        sys.exit(1)
+
+    if not (repo / ".git").is_dir():
+        logger.error(f'"{repo}" has no .git subdir. It should be a Git repository')
+        sys.exit(1)
+
+
+def check_valid_checkout(logger: Logger, repo: Path, checkout_obj: str):
+    results = subprocess.run(
+        ["git", "rev-parse", "--verify", checkout_obj],
+        cwd=str(repo),
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    if not results.returncode == 0:
+        logger.error(f"The string {checkout_obj} was not found in the repository")
+        sys.exit(1)
+
+
+def write_run_log(
+    run_log_path: Path, run_type: str, tag: str, commit: str, config: ConfigParser
+):
+    with run_log_path.open("w") as out_fh:
+        print(f"Run type: {run_type}", file=out_fh)
+        print(f"tag: {tag}", file=out_fh)
+        print(f"Commit: {commit}", file=out_fh)
+
+        print("Config file - settings", file=out_fh)
+        for key, val in config["settings"].items():
+            print(f"{key}: {val}", file=out_fh)
+
+        print(f"Config file - {run_type}", file=out_fh)
+        for key, val in config[run_type].items():
+            print(f"{key}: {val}", file=out_fh)
+
+
+def get_single_csv(
+    config: ConfigParser, run_label: str, run_type: str, start_data: str
+):
+
+    assay = config[run_type]["assay"]
+
+    case_id = config[run_type]["case"]
+    case_dict = config[case_id]
+    case = parse_case(dict(case_dict), start_data, is_trio=False)
+
+    run_csv = CsvEntry(run_label, assay, [case])
+    return run_csv
+
+
+def get_trio_csv(
+    config: ConfigParser,
+    run_label: str,
+    run_type: str,
+    start_data: str,
+):
+
+    assay = config[run_type]["assay"]
+
+    case_ids = config[run_type]["cases"].split(",")
+    assert (
+        len(case_ids) == 3
+    ), f"For a trio, three fields are expected, found: {case_ids}"
+    cases: List[Case] = []
+    for case_id in case_ids:
+        case_dict = config[case_id]
+        case = parse_case(dict(case_dict), start_data, is_trio=True)
+        cases.append(case)
+
+    run_csv = CsvEntry(run_label, assay, cases)
+    return run_csv
+
+
+def parse_case(case_dict: Dict[str, str], start_data: str, is_trio: bool) -> Case:
+    if start_data == "vcf":
+        fw = case_dict["vcf"]
+        rv = case_dict["vcf_tbi"]
+    elif start_data == "bam":
+        fw = case_dict["bam"]
+        rv = case_dict["bam_bai"]
+    else:
+        fw = case_dict["fq_fw"]
+        rv = case_dict["fq_rv"]
```

Comment:
the one suggestion above was to make it easier to add new data types. 

but the other suggestion hidden there was to warn the user before defaulting to fq. 

(though i'd exit with error at this point, as otherwise the default becomes proceeding with something the user did not ask for and potentially a heavier analysis on the cluster that the user might not even need/want)

Reply:
OK, I agree with the point about being strict on the `start_data` type. `fq` is anyway the default in argparse, so if someone is actively messing it up we should indeed crash.

On the generalization ... I think there is a place for simple primitive code. This is the place. It is not many lines, and just needs to be done in this place, and it is easy to understand as is. What do you think? (Happy to discuss!)